### ISSUE TITLE
Logs are rotated from LogAscii::logdir into cwd

### ISSUE THREAD

**Issue by awelzel** [zeek/zeek#2225]
Splitting this out from a comment on #2120.

When setting `LogAscii::logdir` and enabling log rotation, logs are created within `LogAscii::logdir`, but then rotated into the cwd of the Zeek process:

```
$ zeek -r rotation.trace LogAscii::logdir=logs Log::default_rotation_interval=4hr
$ tree
.
├── conn.2011-03-07-04-59-55.log
├── conn.2011-03-07-08-00-05.log
├── conn.2011-03-07-12-00-05.log
├── logs
│   ├── conn.log
│   └── reporter.log
├── packet_filter.2011-03-07-04-00-05.log
└── rotation.trace

1 directory, 7 files
```

Minimally, this seems quite surprising behavior. It appears reasonable to expect rotated logs to be found within `LogAscii::logdir`.

One initial idea for a fix was to prefix any non-absolute rotated path in `Ascii::DoRotate()` with `LogAscii::logdir` if set. But this would change semantics for non-absolute `Log::default_rotation_dir` as it would become relative to `LogAscii::logdir` and also likely confuse the automatic creation of the destination directory.

Alternatively, if `LogAscii::logdir` is set, it may be sensible to use it for the `dir` value of the `RotationPath` returned by `Log::rotation_format_func` unless `Log::default_rotation_dir` is set. It isn't immediately obvious how to get a hold of `LogAscii::logdir` from within `logging/main.zeek` however.

**Comment by rsmmr** [comment]
> Minimally, this seems quite surprising behavior. It appears reasonable to expect rotated logs to be found within `LogAscii::logdir`.

Agree.

> One initial idea for a fix was to prefix any non-absolute rotated path in `Ascii::DoRotate()` with `LogAscii::logdir` if set. But this would change semantics for non-absolute `Log::default_rotation_dir` as it would become relative to `LogAscii::logdir` and also likely confuse the automatic creation of the destination directory.

Yeah, it'd be good not change semantics for `Log::default_rotation_dir`. Also, generally having that relative to CWD, not `logdir`, feels more intuitive to me.
 
> It isn't immediately obvious how to get a hold of `LogAscii::logdir` from within `logging/main.zeek` however.

I think the challenge with the implementation is that the `Log::*` options apply to all writers, but `logdir` is ASCII log specific. I'm wondering if we should elevate `LogAscii::logdir` to `Log::logdir` and make it applicable to any writers writing files (just sqlite actually). Seems that with all rotation stuff in `Log:*`, the `logdir` just fits in there as well.

**Comment by awelzel** [comment]
Closed with #2223.
